packages/make-styles-webpack-loader/src/webpackLoader.test.ts

```diff
@@ -0,0 +1,168 @@
+import * as fs from 'fs';
+import { createFsFromVolume, Volume } from 'memfs';
+import * as path from 'path';
+import * as prettier from 'prettier';
+import * as webpack from 'webpack';
+import { merge } from 'webpack-merge';
+
+async function compileSourceWithWebpack(entryPath: string, configOverrides: webpack.Configuration): Promise<string> {
+  const webpackConfig = merge(
+    {
+      context: __dirname,
+      entry: entryPath,
+
+      mode: 'development',
+
+      output: {
+        path: path.resolve(__dirname),
+        filename: 'bundle.js',
+      },
+
+      module: {
+        rules: [
+          {
+            test: /\.(js|tsx|ts)$/,
+            include: path.dirname(entryPath),
+            use: {
+              loader: path.resolve(__dirname, './index.ts'),
+            },
+          },
+        ],
+      },
+    },
+    configOverrides,
+  );
+  const compiler = webpack(webpackConfig);
+
+  compiler.outputFileSystem = createFsFromVolume(new Volume());
+  compiler.outputFileSystem.join = path.join.bind(path);
+
+  return new Promise((resolve, reject) => {
+    compiler.run((err, stats) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      if (typeof stats === 'undefined') {
+        reject(new Error('"stats" from Webpack are not available, unknown error...'));
+        return;
+      }
+
+      const jsonStats = stats.toJson({ source: true });
+
+      if (stats.hasErrors()) {
+        reject(stats.toJson().errors![0]);
+        return;
+      }
+
+      if (!Array.isArray(jsonStats.modules)) {
+        reject(new Error(`"stats.toJson().modules" should be an array, this could be a compilation error...`));
+        return;
+      }
+
+      const entryModule = jsonStats.modules.find(module => module.nameForCondition === entryPath);
+
+      if (!entryModule) {
+        reject(new Error(`Failed to find a fixture in "stats.toJson().modules", this could be a compilation error...`));
+        return;
+      }
+
+      resolve(entryModule.source as string);
+    });
+  });
+}
+
+function fixLineEndings(value: string) {
+  return String(value).replace(/\r?\n/g, '\n').trim();
+}
+
+/**
+ * Test utility similar to "babel-plugin-tester".
+ *
+ * See https://webpack.js.org/contribute/writing-a-loader/#testing.
+ */
+function testFixture(fixtureName: string, configOverrides: webpack.Configuration = {}) {
```

Comment:
can we provide string union of available fixtures for better DX or throw error early when such a fixture directory doesn't exist outside the `it` block ?

Reply:
- I don't think that typing `fixtureName` with union is a good idea as amount of fixtures will grow once we will discover more cases: Babel plugin has already 23 fixtures
- I also don't see a reason to change conditions as test will fail anyway with a proper warning:

```
Error: Failed to find "code.{js,ts,tsx}" in "office-ui-fabric-react\packages\make-styles-webpack-loader\__fixtures__\object2"
```

Reply:
> I also don't see a reason to change conditions as test will fail anyway with a proper warning:

tbh that message is not explicit, that's why I suggested early exit that will notify the user that such a folder doesnt exist

Reply:
This check will throw if file does not exist, for me this looks enough.